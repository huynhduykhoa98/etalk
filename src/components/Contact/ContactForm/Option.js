/* eslint-disable linebreak-style */
import React, { Component } from 'react';
import { FormattedMessage } from 'react-intl';

class Option extends Component {
  render() {
    return (
      <FormattedMessage
        id={this.props.id}
        values={{ order: `${this.props.value}` }}
      >
        {(message) => <option value={message}>{message}</option>}
      </FormattedMessage>
    );
  }
}

export default Option;
