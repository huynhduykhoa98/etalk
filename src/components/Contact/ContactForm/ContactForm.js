/* eslint-disable linebreak-style */
/* eslint-disable no-console */
/* eslint-disable operator-linebreak */
/* eslint-disable implicit-arrow-linebreak */
import React, { Component } from 'react';
import { Formik, Field, Form } from 'formik';
import { FormattedMessage } from 'react-intl';
import * as Yup from 'yup';
import Swal from 'sweetalert2';
import Option from './Option';

class ContactForm extends Component {
  constructor(props) {
    super(props);
    this.initialValues = {
      txtFullname: '',
      txtEmail: '',
      txtPhoneNumber: '',
      txtCompanyName: '',
      sltProjectType: '',
      sltBudget: '',
      txtDesc: '',
    };
    this.validationSchema = Yup.object().shape({
      txtFullname: Yup.string().required('this field not empty'),
      txtEmail: Yup.string().required('this field not empty'),
      txtPhoneNumber: Yup.number().required('this field not empty'),
      txtCompanyName: Yup.string(),
      sltProjectType: Yup.string(),
      sltBudget: Yup.string(),
      txtDesc: Yup.string(),
    });
  }

  render() {
    return (
      <Formik
        initialValues={this.initialValues}
        validationSchema={this.validationSchema}
        onSubmit={(values) =>
          Swal.fire({ icon: 'success', text: 'OK! Success' }) &&
          console.log(values)
        }
      >
        {(formikProps) => {
          const { errors, touched } = formikProps;
          const showErrortxtFullname =
            errors.txtFullname && touched.txtFullname;
          const showErrortxtEmail = errors.txtEmail && touched.txtEmail;
          const showErrortxtPhoneNumber =
            errors.txtPhoneNumber && touched.txtPhoneNumber;
          const showErrortxtCompanyName =
            errors.txtCompanyName && touched.txtCompanyName;
          const showErrortxtDesc = errors.txtDesc && touched.txtDesc;

          return (
            <Form>
              <fieldset>
                <h3>
                  <FormattedMessage id="contactHeaderInfo" />
                </h3>

                <div className="form-group">
                  <label>
                    <FormattedMessage
                      id="text"
                      defaultMessage="Full Name<strong>*</strong>"
                      values={{
                        strong: (chunk) => <strong>{chunk}</strong>,
                      }}
                    />
                  </label>
                  <Field
                    name="txtFullname"
                    type="text"
                    className="form-control"
                  />
                  {showErrortxtFullname && (
                    <strong>{errors.txtFullname}</strong>
                  )}
                </div>
                <div className="form-group">
                  <label>
                    <FormattedMessage
                      id="text"
                      defaultMessage=" Email Address <strong>*</strong>"
                      values={{
                        strong: (chunk) => <strong>{chunk}</strong>,
                      }}
                    />
                  </label>
                  <Field name="txtEmail" className="form-control" />
                  {showErrortxtEmail && <strong>{errors.txtEmail}</strong>}
                </div>
                <div className="form-group">
                  <label>
                    <FormattedMessage id="text" defaultMessage="Phone Number" />
                  </label>

                  <Field name="txtPhoneNumber" className="form-control" />
                  {showErrortxtPhoneNumber && (
                    <strong>{errors.txtPhoneNumber}</strong>
                  )}
                </div>
                <div className="form-group">
                  <label>
                    <FormattedMessage id="text" defaultMessage="Company Name" />
                  </label>
                  <Field
                    name="txtCompanyName"
                    type="text"
                    className="form-control"
                  />
                  {showErrortxtCompanyName && (
                    <strong>{errors.txtCompanyName}</strong>
                  )}
                </div>
              </fieldset>
              <fieldset>
                <h3>
                  <FormattedMessage id="contactHeaderSkill" />
                </h3>
                <div>
                  <label>
                    <FormattedMessage
                      id="text"
                      defaultMessage="Project Type <strong>*</strong>"
                      values={{
                        strong: (chunk) => <strong>{chunk}</strong>,
                      }}
                    />
                  </label>
                </div>
                <Field
                  as="select"
                  className="custom-select mb-3"
                  style={{ borderColor: '#f90' }}
                  name="sltProjectType"
                >
                  <Option id="projectType" value="1" dat="dat" />

                  <Option id="projectType" value="2" />

                  <Option id="projectType" value="3" />

                  <Option id="projectType" value="4" />

                  <Option id="projectType" value="5" />

                  <Option id="projectType" value="6" />

                  <Option id="projectType" value="7" />

                  <Option id="projectType" value="8" />
                </Field>

                <label>
                  <FormattedMessage
                    id="text"
                    defaultMessage="Estimate Budget <strong>*</strong>"
                    values={{
                      strong: (chunk) => <strong>{chunk}</strong>,
                    }}
                  />
                </label>
                <Field
                  as="select"
                  className="custom-select mb-3"
                  style={{ borderColor: '#f90' }}
                  name="sltBudget"
                >
                  <Option id="selectBudget" value="1" />

                  <Option id="selectBudget" value="2" />

                  <Option id="selectBudget" value="3" />

                  <Option id="selectBudget" value="4" />

                  <Option id="selectBudget" value="5" />

                  <Option id="selectBudget" value="6" />
                </Field>

                <label>
                  <FormattedMessage
                    id="text"
                    defaultMessage="Project Description <strong>*</strong>"
                    values={{
                      strong: (chunk) => <strong>{chunk}</strong>,
                    }}
                  />
                </label>
                <div className="form-group">
                  <Field
                    as="textarea"
                    name="txtDesc"
                    className="form-control textarea"
                    rows={3}
                  />
                  {showErrortxtDesc && <strong>{errors.txtDesc}</strong>}
                </div>
              </fieldset>
              <button type="submit" className="btnsubmit">
                <FormattedMessage id="text" defaultMessage=" REQUEST A QUOTE" />
              </button>
            </Form>
          );
        }}
      </Formik>
    );
  }
}
export default ContactForm;
