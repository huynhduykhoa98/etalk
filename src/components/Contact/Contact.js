/* eslint-disable linebreak-style */
/* eslint-disable class-methods-use-this */
import React, { Component } from 'react';
import { FormattedMessage } from 'react-intl';
import ContactForm from './ContactForm/ContactForm';

class Contact extends Component {
  render() {
    return (
      <section id="contact">
        <div className="container">
          <h2>
            <FormattedMessage id="mainHeader" values={{ order: 6 }} />
          </h2>
          <FormattedMessage
            id="contactText"
            values={{
              p: (...chunks) => <p>{chunks}</p>,
              strong: (...chunks) => <strong>{chunks}</strong>,
            }}
          />
          <ContactForm></ContactForm>
        </div>
      </section>
    );
  }
}

export default Contact;
