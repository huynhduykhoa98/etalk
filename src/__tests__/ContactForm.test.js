/* eslint-disable linebreak-style */
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ContactForm from '../components/Contact/ContactForm/ContactForm';
import Contact from '../components/Contact/Contact';

Enzyme.configure({ adapter: new Adapter() });
// test prop
test('render a document title (prop)', () => {
  const wrapper = shallow(<ContactForm prop="Events" />);
  expect(wrapper.prop('prop')).toEqual('Events');
});
// test shallow
it('test render', () => {
  const wrapped = shallow(<Contact />);
  expect(wrapped.find(ContactForm).length).toEqual(1);
});
