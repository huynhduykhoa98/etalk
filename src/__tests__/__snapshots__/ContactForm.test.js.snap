// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pass a selected value to the onChange handler 1`] = `ShallowWrapper {}`;
